lib/logstash/filters/kv.rb

```diff
@@ -41,29 +41,65 @@ class LogStash::Filters::KV < LogStash::Filters::Base
   # These characters form a regex character class and thus you must escape special regex
   # characters like `[` or `]` using `\`.
   #
-  # For example, to strip `<`, `>`, `[`, `]` and `,` characters from values:
+  # Only leading and trailing characters are trimed from the value.
+  #
+  # For example, to trim `<`, `>`, `[`, `]` and `,` characters from values:
   # [source,ruby]
   #     filter {
   #       kv {
-  #         trim => "<>\[\],"
+  #         trim_value => "<>\[\],"
   #       }
   #     }
-  config :trim, :validate => :string
```

Comment:
Hindsight:

THis should have been kept as a setting, but adding `:deprecated => "This setting has been renamed to 'trim_value'."`